packages/core/src/model.ts

```diff
@@ -291,8 +292,15 @@ export class ChartModel {
 	 * @param newOptions New options to be set
 	 */
 	setOptions(newOptions) {
+		const mergedOptions = Tools.merge(this.getOptions(), newOptions);
+		// these options need assignment, not just merge
+		if (Tools.getProperty(newOptions, "zoomBar", "top", "data")) {
+			mergedOptions.zoomBar.top.data = Tools.clone(
+				newOptions.zoomBar.top.data
+			);
+		}
```

Comment:
why is this needed?

Reply:
If use set new zoom bar data, we should not merge the new data to old data. We should replace old data with new data. The `Tools.merge` will merge new data to old data, and that's not expected.